internal/parser/parser.go

```diff
@@ -622,6 +622,8 @@ func (p *parser) parseObjectRemainder(tok *token) (ast.Node, *token, error) {
 		if next.kind == tokenComma {
 			gotComma = true
 			next = p.pop()
+		} else {
+			gotComma = false
```

Comment:
Wow, that variable is defined 268 lines up from here. It's easy to lose track of where it's going to be consulted and reset.

Reply:
It'd probably benefit from splitting the cases and the ForSpec stuff out into functions, but the next token and gotComma has to be passed into all of them and some additional other variables as well in some cases

Reply:
I was working on this but then the HDD I was doing it on died so I'll probably have to do it again :)  But it wasn't hard.  I need my new laptop to arrive first though probably.

Reply:
Finally re-did it #335